File: 프로그래머스/1/388351. 유연근무제/유연근무제.java
import java.util.*;

class Solution {
    public int solution(int[] schedules, int[][] timelogs, int startday) {
        int[] day = new int[7];
        int idx = 0;
        for (int i = startday - 1; i < startday + 7 - 1; i++) {
            day[idx++] = i % 7 + 1;
        }
        
        int[] limit = new int[schedules.length];
        for (int i = 0; i < schedules.length; i++) {
            if ((schedules[i] + 10) % 100 < 60) {
                limit[i] = schedules[i] + 10;
            } else {
                int div = (int) (schedules[i] + 10) / 100;
                int mod = (int) (schedules[i] + 10) % 100;
                limit[i] = (div + 1) * 100 + (mod % 60);
            }
        }

        int answer = 0;
        for (int i = 0; i < timelogs.length; i++) {
            boolean check = false;
            for (int j = 0; j < 7; j++) {
                if (day[j] == 6 || day[j] == 7) continue;
                if (timelogs[i][j] > limit[i]) {
                    check = true;
                    break;
                }
            }
            if (!check) answer ++; 
        }
        return answer;
    }
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.py
def solution(schedules, timelogs, startday):
    day = [i % 7 + 1 for i in range(startday - 1, startday + 7 - 1)]
    limit_schedules = []
    for schedule in schedules:
        if (schedule + 10) % 100 < 60:
            limit_schedules.append(schedule + 10)
        else:
            div, mod = divmod(schedule + 10, 100)
            limit_schedules.append((div + 1) * 100 + (mod % 60))

    answer = 0
    for i, timelog in enumerate(timelogs):
        for d, t in zip(day, timelog):
            if d in [6, 7]:
                continue
            if t > limit_schedules[i]:
                break
        else:
            answer += 1
    return answer